src/include/shared/sgxlkl_enclave_config_gen.h

```diff
@@ -0,0 +1,148 @@
+#ifndef SGXLKL_ENCLAVE_CONFIG_GEN_H
+#define SGXLKL_ENCLAVE_CONFIG_GEN_H
+
+/* Automatically generated from ../../tools/schemas/enclave-config.schema.json; do not modify. */
+
+#include <inttypes.h>
+#include <stdbool.h>
+#include <stddef.h>
+#include <elf.h>
+
+#define SGXLKL_ENCLAVE_CONFIG_VERSION 1UL
+
+typedef enum
+{
+    UNKNOWN_MODE = 0,
+    SW_DEBUG_MODE = 1,
+    HW_DEBUG_MODE = 2,
+    HW_RELEASE_MODE = 3
+} sgxlkl_enclave_mode_t;
+
+const char* sgxlkl_enclave_mode_t_to_string(sgxlkl_enclave_mode_t e);
+sgxlkl_enclave_mode_t string_to_sgxlkl_enclave_mode_t(const char *e);
+
+typedef enum
+{
+    ENCLAVE_MMAP_FILES_NONE = 0,
+    ENCLAVE_MMAP_FILES_PRIVATE = 1,
+    ENCLAVE_MMAP_FILES_SHARED = 2
+} sgxlkl_enclave_mmap_files_t;
+
+const char* sgxlkl_enclave_mmap_files_t_to_string(sgxlkl_enclave_mmap_files_t e);
+sgxlkl_enclave_mmap_files_t string_to_sgxlkl_enclave_mmap_files_t(const char *e);
+
+typedef enum
+{
+    EXIT_STATUS_FULL = 0,
+    EXIT_STATUS_BINARY = 1,
+    EXIT_STATUS_NONE = 2
+} sgxlkl_exit_status_mode_t;
+
+const char* sgxlkl_exit_status_mode_t_to_string(sgxlkl_exit_status_mode_t e);
+sgxlkl_exit_status_mode_t string_to_sgxlkl_exit_status_mode_t(const char *e);
+
+typedef struct sgxlkl_clock_res_config
+{
+    char resolution[17];
+} sgxlkl_clock_res_config_t;
+
+typedef struct sgxlkl_enclave_mount_config
+{
+    bool create;
+    char destination[256];
+    size_t key_len;
+    uint8_t* key;
+    char* key_id;
+    bool fresh_key;
+    bool readonly;
+    char* roothash;
+    size_t roothash_offset;
+    size_t size;
+} sgxlkl_enclave_mount_config_t;
+
+typedef struct sgxlkl_enclave_root_config
+{
+    size_t key_len;
+    uint8_t* key;
+    char* key_id;
+    char* roothash;
+    size_t roothash_offset;
+    bool readonly;
+    bool overlay;
+} sgxlkl_enclave_root_config_t;
+
+typedef struct sgxlkl_enclave_wg_peer_config
+{
+    char* key;
+    char* allowed_ips;
```

Comment:
`uint32_t *`?

Reply:
No. Never was a `uint32_t*` and it would needlessly complicate things. The semantics are indeed `char*`, i.e. this is a null-terminated string containing ASCII characters.

Reply:
Future-compatible with IPv6! :)

Reply:
Hah! Good point :-)